packages/contentstack-branch/src/commands/cm/branches/index.ts

```diff
@@ -1,24 +1,81 @@
 import { Command } from '@contentstack/cli-command';
-import { cliux, messageHandler, printFlagDeprecation, managementSDKClient, flags } from '@contentstack/cli-utilities';
+import { cliux, messageHandler, managementSDKClient, flags, configHandler } from '@contentstack/cli-utilities';
+import { getbranchesList } from '../../../utils/index';
+import chalk from 'chalk';
+import { existsSync, readFileSync } from 'fs';
 
 export default class BranchListCommand extends Command {
   static description: string = messageHandler.parse('List the branches to select'); // Note: Update the description
 
-  static examples: string[] = ['csdx cm:branches --base-branch main --stack-api-key bltxxxxxxxx']; // Note: Add and modify the examples
+  static examples: string[] = ['csdx cm:branches', 'csdx cm:branches --verbose']; // Note: Add and modify the examples
 
-  static usage: string = 'cm:branches [--base-branch <value>] [--stack-api-key <value>]'; // Note: Add and modify the usage
+  static usage: string = 'cm:branches'; // Note: Add and modify the usage
 
-  static flags = {};
+  static flags = {
+    verbose: flags.boolean({ char: 'v', description: 'Verbose' }),
+  };
 
   static aliases: string[] = []; // Note: alternative usage if any
 
   async run(): Promise<any> {
     try {
       const managementAPIClient = await managementSDKClient({ host: this.cmaHost });
       const { flags: branchListFlags } = await this.parse(BranchListCommand);
-      /**
-       * Implementation
-       */
-    } catch (error) {}
+      let verbose = branchListFlags['verbose'];
+      let config;
+
+      if (existsSync(__dirname + '/branch-config.json')) {
+        let data = JSON.parse(readFileSync(__dirname + '/branch-config.json', 'utf-8'));
+        config = data['base-branch'];
+      } else if (Boolean(configHandler.get(`base-branch`))) {
+        config = configHandler.get(`base-branch`);
+      }
+
+      if (config && Object.keys(config).length > 0) {
+        const branchResult = await managementAPIClient.stack({ api_key: config.apiKey }).branch().query().find();
```

Comment:
if the config not set this code will print branch is not available, which is wrong. Expectation is it needs to fetch the branch list always and to identify the current base branch logic needs to make use of the config.